aiida/manage/configuration/__init__.py

```diff
@@ -59,6 +59,24 @@
 BACKEND_UUID = None  # This will be set to the UUID of the profile as soon as its corresponding backend is loaded
 
 
+def check_rabbitmq_version():
+    """Check the version of RabbitMQ that is being connected to and emit warning if the version is not compatible.
+
+    Versions 3.8 and above are not compatible with AiiDA with default configuration.
+    """
+    from packaging.version import parse
+
+    from aiida.cmdline.utils import echo
+    from aiida.manage.manager import get_manager
+
+    url = 'https://github.com/aiidateam/aiida-core/wiki/RabbitMQ-version-to-use'
+    communicator = get_manager().get_communicator()
+    version = parse(communicator.server_properties['version'].decode('utf-8'))
+    if version >= parse('3.8'):
+        echo.echo_warning(f'Connected to RabbitMQ v{version} which is known to cause problems)')
+        echo.echo_warning(f'See {url} for details.')
```

Comment:
Is it OK to use the `cmdline` module functions like `echo` outside of it? Maybe this function should just return True/Fals, and the warning be issued elsewhere.

Reply:
Here it is I feel. The `echo` commands essentially just directly use the `logging` module and when invoked through `verdi` it will simply be configured to write to stdout. When I refactored it to be that way, I decided to just not remove it from the `aiida.cmdline` package since I know other packages were using it as well and so it would break. That being said, it probably should not be used everywhere else internally.

Reply:
Ok, I see. I think it would still be good to avoid directly importing from there - maybe move the function in some utils module, and keep the aiida.cmdline for backward compatibility, proxying to the utils function? Anyway, just a suggestion, not directly related to this issue

Reply:
I think it should actually stay in the `aiida.cmdline` module, because that is what it was designed for. So far it is only used in migrations in `aiida.backend` and in the `aiida.manage` module, which I think is acceptable. However, if we want to improve this, we should probably just look into having these modules use the normal `logging.warning` and equivalent and then make sure that their formatters are set to work just as the one for the command line logger. This way the log messages will be formatted uniformly. Will open an issue for this.